pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -466,27 +319,42 @@ abstract contract YieldBearingPoolSwapBase is BaseVaultTest {
         SwapResultLocals memory vars = _createSwapResultLocals(
             SwapKind.EXACT_IN,
             IERC4626(address(paths[0].steps[0].tokenOut)),
-            IERC4626(address(paths[0].steps[1].tokenOut))
+            IERC4626(address(paths[0].steps[1].tokenOut)),
+            withBufferLiquidity
         );
-        vars.expectedDeltaTokenIn = paths[0].exactAmountIn;
+
+        vars.expectedUnderlyingDeltaTokenIn = paths[0].exactAmountIn;
 
         if (paths[0].tokenIn == _token1Fork) {
-            uint256 wrappedAmountIn = _previewDepositWithBuffer(ybToken1, vars.expectedDeltaTokenIn, shouldUseConvert);
-            uint256 wrappedAmountInScaled18 = wrappedAmountIn.divDown(_ybToken1Factor);
+            (
+                vars.expectedWrappedDeltaTokenIn,
+                vars.expectedUnderlyingSurplusTokenIn,
+                vars.expectedWrappedSurplusTokenIn
+            ) = _previewDepositWithBuffer(ybToken1, vars.expectedUnderlyingDeltaTokenIn, withBufferLiquidity);
+            uint256 wrappedAmountInScaled18 = vars.expectedWrappedDeltaTokenIn.divDown(_ybToken1Factor);
             // PoolMock is linear, so wrappedAmountInScaled18 = wrappedAmountOutScaled18
-            uint256 wrappedAmountOutRaw = wrappedAmountInScaled18.mulDown(_ybToken2Factor);
-            vars.expectedDeltaTokenOut = _previewRedeemWithBuffer(ybToken2, wrappedAmountOutRaw, shouldUseConvert);
+            vars.expectedWrappedDeltaTokenOut = wrappedAmountInScaled18.mulDown(_ybToken2Factor);
+            (
+                vars.expectedUnderlyingDeltaTokenOut,
+                vars.expectedUnderlyingSurplusTokenOut,
+                vars.expectedWrappedSurplusTokenOut
+            ) = _previewRedeemWithBuffer(ybToken2, vars.expectedWrappedDeltaTokenOut, withBufferLiquidity);
```

Comment:
So if I understand this correctly, now we also have expected surplus for wrapped and underlying (one for buffer in, one for buffer out). Is that correct?

Reply:
We were calculating the surplus before elsewhere, in the test function, and this calculation was different depending n the case. Now the expected surplus is standard.